src/FsAutoComplete.Core/Commands.fs

```diff
@@ -457,3 +457,54 @@ type Commands (serialize : Serializer) =
 
         return [Response.workspacePeek serialize d]
     }
+
+    member x.GetUnusedDeclarations file =
+        let file = Path.GetFullPath file
+        let isScript = file.EndsWith ".fsx"
+
+        async {
+            match state.TryGetFileCheckerOptionsWithSource file with
+            | Failure s ->  return [Response.error serialize s]
+            | Success (opts, _) ->
+                let tyResOpt = checker.TryGetRecentCheckResultsForFile(file, opts)
+                match tyResOpt with
+                | None -> return [ Response.info serialize "Cached typecheck results not yet available"]
+                | Some tyRes ->
+                    let! allUses = tyRes.GetCheckResults.GetAllUsesOfAllSymbolsInFile ()
+                    let unused = UnusedDeclarationsAnalyzer.getUnusedDeclarationRanges allUses isScript
+                    return [ Response.unusedDeclarations serialize unused ]
+        } |> x.AsCancellable file
+
+    member x.GetSimplifiedNames file =
+        let file = Path.GetFullPath file
```

Comment:
we need a safe full path.

Reply:
`Path.GetFullPath` is used all over the place in FSAC. Also, I believe that in most cases we assume that paths sent by a client are absolute, we just use `Path.GetFullPath` as a way to normalize them (make sure same slashes are used everywhere and get rid of things like `\..\`

Reply:
make sense.